provider/agent_test.go

```diff
@@ -241,65 +243,166 @@ func TestAgent_ResourcesMonitoring(t *testing.T) {
+					require.Len(t, state.Modules[0].Resources, 1)
+
+					resource := state.Modules[0].Resources["coder_agent.dev"]
+					require.NotNil(t, resource)
+
+					attr := resource.Primary.Attributes
+					require.Equal(t, "1", attr["resources_monitoring.#"])
+					require.Equal(t, "1", attr["resources_monitoring.0.memory.#"])
+					require.Equal(t, "80", attr["resources_monitoring.0.memory.0.threshold"])
+					return nil
+				},
+			}},
+		})
+	})
+	t.Run("MultipleMemory", func(t *testing.T) {
+		resource.Test(t, resource.TestCase{
+			ProviderFactories: coderFactory(),
+			IsUnitTest:        true,
+			Steps: []resource.TestStep{{
+				Config: `
+					provider "coder" {
+						url = "https://example.com"
 					}
-				}`,
-			Check: func(state *terraform.State) error {
-				require.Len(t, state.Modules, 1)
-				require.Len(t, state.Modules[0].Resources, 1)
+					resource "coder_agent" "dev" {
+						os = "linux"
+						arch = "amd64"
+						resources_monitoring {
+							memory {
+								enabled = true
+								threshold = 80
+							}
+							memory {
+								enabled = true
+								threshold = 90
+							}
+						}
+					}`,
+				ExpectError: regexp.MustCompile(`No more than 1 "memory" blocks are allowed`),
+			}},
+		})
+	})
 
-				resource := state.Modules[0].Resources["coder_agent.dev"]
-				require.NotNil(t, resource)
+	t.Run("InvalidThreshold", func(t *testing.T) {
+		resource.Test(t, resource.TestCase{
+			ProviderFactories: coderFactory(),
+			IsUnitTest:        true,
+			Steps: []resource.TestStep{{
+				Config: `
+					provider "coder" {
+						url = "https://example.com"
+					}
+					resource "coder_agent" "dev" {
+						os = "linux"
+						arch = "amd64"
+						resources_monitoring {
+							memory {
+								enabled = true
+								threshold = 101
+							}
+						}
+					}`,
+				ExpectError: regexp.MustCompile("threshold must be between 0 and 100"),
+			}},
+		})
+	})
 
-				t.Logf("resource: %v", resource.Primary.Attributes)
+	t.Run("DuplicatePaths", func(t *testing.T) {
+		resource.Test(t, resource.TestCase{
+			ProviderFactories: coderFactory(),
+			IsUnitTest:        true,
+			Steps: []resource.TestStep{{
+				Config: `
+					provider "coder" {
+						url = "https://example.com"
+					}
+					resource "coder_agent" "dev" {
+						os = "linux"
+						arch = "amd64"
+						resources_monitoring {
+							volume {
+								path = "/volume1"
+								enabled = true
+								threshold = 80
+							}
+							volume {
+								path = "/volume1"
+								enabled = true
+								threshold = 100
+							}
+						}
+					}`,
+				ExpectError: regexp.MustCompile("duplicate volume path"),
+			}},
+		})
+	})
 
-				attr := resource.Primary.Attributes
-				require.Equal(t, "1", attr["resources_monitoring.#"])
-				require.Equal(t, "1", attr["resources_monitoring.0.memory.#"])
-				require.Equal(t, "80", attr["resources_monitoring.0.memory.0.threshold"])
-				return nil
-			},
-		}},
+	t.Run("NoPath", func(t *testing.T) {
```

Comment:
These are good 👍 
You're missing too cases, though: empty volume and non-absolute path volume.

Reply:
Nice 👍 thanks